### PR TITLE
[APP-2727] - Implement Editorials Logic

### PR DESCRIPTION
**Pull request by GuiFPires**
**What does this PR do?**

It implements the whole logic of Editorials in home. It also refactors some urls and edits already made views in order for them to match the design in figma.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] AppViewScreen.kt
- [ ] RepositoryModule.kt
- [ ] EditorialMoreView.kt
- [ ] EditorialView.kt
- [ ] EditorialViewScreen.kt
- [ ] MainView.kt
- [ ] WidgetTypeJSON.kt
- [ ] BundlesUseCase.kt
- [ ] Widget.kt
- [ ] ArticlesMetaUseCase.kt
- [ ] Article.kt
- [ ] ArticleListUiState.kt
- [ ] EditorialsListViewModel.kt
- [ ] ViewModelProvider.kt

**How should this be manually tested?**

  Test to see if all the editorial urls work without the widget in home and if they match the design. 
  Test to see if all editorial views (including in home) work with the widget in home. While we don't have the widget in home, you can use the charles rewrite left in the ticket to test.
  
  Flow on how to test this or QA Tickets related to this use-case: [AND-49](https://aptoide.atlassian.net/browse/AND-49)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-49](https://aptoide.atlassian.net/browse/AND-49)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[AND-49]: https://aptoide.atlassian.net/browse/AND-49?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialViewScreen.kt [hunk 1]
```diff
@@ -259,12 +262,10 @@ private fun ArticleViewContent(
         }
         item {
           EditorialBundle(
-            modifier = Modifier
-              .fillMaxWidth()
-              .background(color = Palette.Black),
+            modifier = Modifier.padding(top = 16.dp),
```
Comment (Culurio) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialViewScreen.kt:265]: Acording to figma the top padding should be 40.dp

Comment (GuiFPires) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialViewScreen.kt]: Fixed. It is now 16 of the bottom padding of the text + 24 of the top padding of the bundle

**Review by Culurio — APPROVED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialMoreView.kt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    modifier = modifier,
+    text = articleMeta.summary,
+    maxLines = 3,
+    overflow = TextOverflow.Ellipsis,
+    style = AGTypography.SmallGames,
+    color = Palette.White
+  )
+}
+
+fun buildSeeMoreEditorialsRoute(
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialMoreView.kt:204]: Usually we keep it closer to the beginning of the file

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/di/RepositoryModule.kt [hunk 3]
```diff
@@ -64,7 +64,7 @@ class RepositoryModule {
   @Provides
   @DefaultEditorialUrl
   fun provideDefaultEditorialUrl(): String =
-    "${BuildConfig.STORE_DOMAIN}user/action/item/cards/get/type=CURATION_1/limit=10/store_name=${BuildConfig.MARKET_NAME}/aptoide_uid=1/?subtype=COLLECTION&aab=1&aptoide_vercode=${BuildConfig.VERSION_CODE}"
+    "${BuildConfig.STORE_DOMAIN}user/action/item/cards/get/type=CURATION_1/limit=10/store_name=${BuildConfig.MARKET_NAME}"
```
[the remark this answers is not in the document]
Comment (GuiFPires) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/di/RepositoryModule.kt:67]: uid and vercode are already sent alongside with the q, so it doesn't make sense to send it twice
subtype=COLLECTION is wrong and shouldn't be sent, I don't understand why it was sent in the first place

File: app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialViewScreen.kt [hunk 4]
```diff
@@ -259,12 +262,10 @@ private fun ArticleViewContent(
...
         item {
           EditorialBundle(
-            modifier = Modifier
-              .fillMaxWidth()
-              .background(color = Palette.Black),
+            modifier = Modifier.padding(top = 24.dp),
             bundle = Bundle(
               title = stringResource(R.string.editorial_more_articles_title),
-              actions = emptyList(),
+              actions = listOf(WidgetAction(type = BUTTON, tag = "editorial-more", url = "null")),
```
[the remark this answers is not in the document]
Comment (GuiFPires) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/editorial/EditorialViewScreen.kt:268]: Yes, for the more articles.

File: feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/ArticleListUiState.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+sealed class ArticleListUiState {
+  data class Idle(val articles: List<ArticleMeta>) : ArticleListUiState()
+  object Loading : ArticleListUiState()
+  object Empty : ArticleListUiState()
+  object NoConnection : ArticleListUiState()
+  object Error : ArticleListUiState()
+}
+
+class ArticleListUiStateProvider : PreviewParameterProvider<ArticleListUiState> {
```
[the remark this answers is not in the document]
Comment (GuiFPires) [reply feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/presentation/ArticleListUiState.kt:17]: I followed the logic that we had on AppsListUiState, so we can have preview screens with every state

**Review by G0retZ — APPROVED** [review]

**Pull request merged**